MAINT: mypy failing in CI

https://github.com/scipy/scipy/actions/runs/27759777389/job/82130842332?pr=25400
```
scipy/sparse/tests/test_construct.py:935: error: List item 0 has incompatible type "def diags_array(Any, /, *, offsets: Any = ..., shape: Any = ..., format: Any = ..., dtype: Any = ...) -> Any"; expected "Callable[[Any, Any, Any, Any, Any], Any]"  [list-item]
scipy/special/tests/test_orthogonal.py:409: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float", "int", "int"  [call-overload]
scipy/special/tests/test_orthogonal.py:409: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:409: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:409: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:409: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:409: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:464: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float", "int", "int"  [call-overload]
scipy/special/tests/test_orthogonal.py:464: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:464: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:464: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:464: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:464: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:495: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:495: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:495: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:495: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:495: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:495: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:544: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:544: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:544: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:544: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:544: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:544: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:607: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float", "int"  [call-overload]
scipy/special/tests/test_orthogonal.py:607: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:607: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:607: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:607: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:607: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:626: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:626: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:626: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:626: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:626: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:626: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:649: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:649: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:649: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:649: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:649: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:649: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:667: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:667: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:667: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:667: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:667: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:667: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:684: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:684: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:684: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:684: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:684: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:684: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:702: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:702: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:702: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:702: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:702: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:702: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:720: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:720: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:720: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:720: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:720: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:720: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:739: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:739: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:739: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:739: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:739: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:739: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:758: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:758: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:758: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:758: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:758: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:758: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:777: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float"  [call-overload]
scipy/special/tests/test_orthogonal.py:777: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:777: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:777: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:777: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:777: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/special/tests/test_orthogonal.py:817: error: No overload variant of "raises" matches argument types "type[ValueError]", overloaded function, "float", "int"  [call-overload]
scipy/special/tests/test_orthogonal.py:817: note: Possible overload variants:
scipy/special/tests/test_orthogonal.py:817: note:     def [E: BaseException] raises(expected_exception: type[E] | tuple[type[E], ...], *, match: str | Pattern[str] | None = ..., check: Callable[[E], bool] = ...) -> RaisesExc[E]
scipy/special/tests/test_orthogonal.py:817: note:     def raises(*, match: str | Pattern[str], check: Callable[[BaseException], bool] = ...) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:817: note:     def raises(*, check: Callable[[BaseException], bool]) -> RaisesExc[BaseException]
scipy/special/tests/test_orthogonal.py:817: note:     def [E: BaseException, P`-2] raises(expected_exception: type[E] | tuple[type[E], ...], func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> ExceptionInfo[E]
scipy/interpolate/tests/test_fitpack2.py:1423: error: List item 1 has incompatible type "tuple[def _regrid(x: Any, y: Any, z: Any, *, bbox: Any = ..., kx: Any = ..., ky: Any = ..., s: Any = ..., maxit: Any = ...) -> Any, Callable[[Any, Any, Any, Any, Any, Any], Any]]"; expected "tuple[Callable[[Any, Any, Any, Any, Any, Any, Any, Any], Any], function]"  [list-item]
scipy/interpolate/tests/test_fitpack2.py:1452: error: List item 1 has incompatible type "tuple[def _regrid(x: Any, y: Any, z: Any, *, bbox: Any = ..., kx: Any = ..., ky: Any = ..., s: Any = ..., maxit: Any = ...) -> Any, Callable[[Any, Any, Any, Any, Any, Any], Any]]"; expected "tuple[Callable[[Any, Any, Any, Any, Any, Any, Any, Any], Any], function]"  [list-item]
scipy/interpolate/tests/test_fitpack2.py:1477: error: List item 1 has incompatible type "tuple[def _regrid(x: Any, y: Any, z: Any, *, bbox: Any = ..., kx: Any = ..., ky: Any = ..., s: Any = ..., maxit: Any = ...) -> Any, Callable[[Any, Any, Any, Any, Any, Any], Any]]"; expected "tuple[Callable[[Any, Any, Any, Any, Any, Any, Any, Any], Any], function]"  [list-item]
```

--- a/scipy/interpolate/tests/test_fitpack2.py
+++ b/scipy/interpolate/tests/test_fitpack2.py
@@ -1418,10 +1418,10 @@ class TestRectBivariateSpline:
     @pytest.mark.parametrize('s_tols', [(0, 1e-12, 1e-7),
                                         (1, 7e-3, 1e-4),
                                         (3, 2e-2, 1e-4)])
-    @pytest.mark.parametrize("spl_apis", [(RectBivariateSpline,
+    @pytest.mark.parametrize("spl_apis", ((RectBivariateSpline,
                                            _RectBivariateSplineEval),
                                           (_regrid,
-                                           _ndbspline_call_like_bivariate)])
+                                           _ndbspline_call_like_bivariate)))
     def test_spline_large_2d(self, shape, s_tols, spl_apis):
         # Reference - https://github.com/scipy/scipy/issues/17787
         #
@@ -1447,10 +1447,10 @@ class TestRectBivariateSpline:
     @pytest.mark.skipif(sys.maxsize <= 2**32, reason="Segfaults on 32-bit system "
                                                      "due to large input data")
     @pytest.mark.parametrize("k", [3, 4])
-    @pytest.mark.parametrize("spl_apis", [(RectBivariateSpline,
+    @pytest.mark.parametrize("spl_apis", ((RectBivariateSpline,
                                            _RectBivariateSplineEval),
                                           (_regrid,
-                                           _ndbspline_call_like_bivariate)])
+                                           _ndbspline_call_like_bivariate)))
     def test_spline_large_2d_maxit(self, k, spl_apis):
         # Reference - for https://github.com/scipy/scipy/issues/17787
         #
@@ -1472,10 +1472,10 @@ class TestRectBivariateSpline:
         assert(not np.isnan(z_spl).any())
         xp_assert_close(z_spl, z, atol=atol, rtol=rtol)
 
-    @pytest.mark.parametrize("spl_apis", [(RectBivariateSpline,
+    @pytest.mark.parametrize("spl_apis", ((RectBivariateSpline,
                                            _RectBivariateSplineEval),
                                           (_regrid,
-                                           _ndbspline_call_like_bivariate)])
+                                           _ndbspline_call_like_bivariate)))
     def test_spline_synthetic_data(self, spl_apis):
         """
         Test regrid with synthetic smooth data (mixed frequencies + noise).

--- a/scipy/sparse/tests/test_construct.py
+++ b/scipy/sparse/tests/test_construct.py
@@ -932,7 +932,7 @@ def test_diags_array():
         construct.diags(np.arange(1.0, 5.0), 5, shape=(4, 4))
 
 
-@pytest.mark.parametrize('func', [construct.diags_array, construct.diags])
+@pytest.mark.parametrize('func', (construct.diags_array, construct.diags))
 def test_diags_int(func):
     d = [[3], [1, 2], [4]]
     offsets = [-1, 0, 1]

--- a/scipy/special/tests/test_orthogonal.py
+++ b/scipy/special/tests/test_orthogonal.py
@@ -406,7 +406,7 @@ def test_roots_jacobi():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_jacobi, 0, 1, 1)
-    assert_raises(ValueError, sc.roots_jacobi, 3.3, 1, 1)
+    assert_raises(ValueError, sc.roots_jacobi, 3.3, 1, 1)  # type: ignore[call-overload]
     assert_raises(ValueError, sc.roots_jacobi, 3, -2, 1)
     assert_raises(ValueError, sc.roots_jacobi, 3, 1, -2)
     assert_raises(ValueError, sc.roots_jacobi, 3, -2, -2)
@@ -461,7 +461,7 @@ def test_roots_sh_jacobi():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_sh_jacobi, 0, 1, 1)
-    assert_raises(ValueError, sc.roots_sh_jacobi, 3.3, 1, 1)
+    assert_raises(ValueError, sc.roots_sh_jacobi, 3.3, 1, 1)  # type: ignore[call-overload]
     assert_raises(ValueError, sc.roots_sh_jacobi, 3, 1, 2)    # p - q <= -1
     assert_raises(ValueError, sc.roots_sh_jacobi, 3, 2, -1)   # q <= 0
     assert_raises(ValueError, sc.roots_sh_jacobi, 3, -2, -1)  # both
@@ -492,7 +492,7 @@ def test_roots_hermite():
     assert_allclose(sum(v), m, 1e-14, 1e-14)
 
     assert_raises(ValueError, sc.roots_hermite, 0)
-    assert_raises(ValueError, sc.roots_hermite, 3.3)
+    assert_raises(ValueError, sc.roots_hermite, 3.3)  # type: ignore[call-overload]
 
 def test_roots_hermite_asy():
     # Recursion for Hermite functions
@@ -541,7 +541,7 @@ def test_roots_hermitenorm():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_hermitenorm, 0)
-    assert_raises(ValueError, sc.roots_hermitenorm, 3.3)
+    assert_raises(ValueError, sc.roots_hermitenorm, 3.3)  # type: ignore[call-overload]
 
 def test_roots_gegenbauer():
     def rootf(a):
@@ -604,7 +604,7 @@ def test_roots_gegenbauer():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_gegenbauer, 0, 2)
-    assert_raises(ValueError, sc.roots_gegenbauer, 3.3, 2)
+    assert_raises(ValueError, sc.roots_gegenbauer, 3.3, 2)  # type: ignore[call-overload]
     assert_raises(ValueError, sc.roots_gegenbauer, 3, -.75)
 
 def test_roots_chebyt():
@@ -623,7 +623,7 @@ def test_roots_chebyt():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_chebyt, 0)
-    assert_raises(ValueError, sc.roots_chebyt, 3.3)
+    assert_raises(ValueError, sc.roots_chebyt, 3.3)  # type: ignore[call-overload]
 
 def test_chebyt_symmetry():
     x, w = sc.roots_chebyt(21)
@@ -646,7 +646,7 @@ def test_roots_chebyu():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_chebyu, 0)
-    assert_raises(ValueError, sc.roots_chebyu, 3.3)
+    assert_raises(ValueError, sc.roots_chebyu, 3.3)  # type: ignore[call-overload]
 
 def test_roots_chebyc():
     weightf = orth.chebyc(5).weight_func
@@ -664,7 +664,7 @@ def test_roots_chebyc():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_chebyc, 0)
-    assert_raises(ValueError, sc.roots_chebyc, 3.3)
+    assert_raises(ValueError, sc.roots_chebyc, 3.3)  # type: ignore[call-overload]
 
 def test_roots_chebys():
     weightf = orth.chebys(5).weight_func
@@ -681,7 +681,7 @@ def test_roots_chebys():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_chebys, 0)
-    assert_raises(ValueError, sc.roots_chebys, 3.3)
+    assert_raises(ValueError, sc.roots_chebys, 3.3)  # type: ignore[call-overload]
 
 def test_roots_sh_chebyt():
     weightf = orth.sh_chebyt(5).weight_func
@@ -699,7 +699,7 @@ def test_roots_sh_chebyt():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_sh_chebyt, 0)
-    assert_raises(ValueError, sc.roots_sh_chebyt, 3.3)
+    assert_raises(ValueError, sc.roots_sh_chebyt, 3.3)  # type: ignore[call-overload]
 
 def test_roots_sh_chebyu():
     weightf = orth.sh_chebyu(5).weight_func
@@ -717,7 +717,7 @@ def test_roots_sh_chebyu():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_sh_chebyu, 0)
-    assert_raises(ValueError, sc.roots_sh_chebyu, 3.3)
+    assert_raises(ValueError, sc.roots_sh_chebyu, 3.3)  # type: ignore[call-overload]
 
 def test_roots_legendre():
     weightf = orth.legendre(5).weight_func
@@ -736,7 +736,7 @@ def test_roots_legendre():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_legendre, 0)
-    assert_raises(ValueError, sc.roots_legendre, 3.3)
+    assert_raises(ValueError, sc.roots_legendre, 3.3)  # type: ignore[call-overload]
 
 def test_roots_sh_legendre():
     weightf = orth.sh_legendre(5).weight_func
@@ -755,7 +755,7 @@ def test_roots_sh_legendre():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_sh_legendre, 0)
-    assert_raises(ValueError, sc.roots_sh_legendre, 3.3)
+    assert_raises(ValueError, sc.roots_sh_legendre, 3.3)  # type: ignore[call-overload]
 
 def test_roots_laguerre():
     weightf = orth.laguerre(5).weight_func
@@ -774,7 +774,7 @@ def test_roots_laguerre():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_laguerre, 0)
-    assert_raises(ValueError, sc.roots_laguerre, 3.3)
+    assert_raises(ValueError, sc.roots_laguerre, 3.3)  # type: ignore[call-overload]
 
 def test_roots_genlaguerre():
     def rootf(a):
@@ -814,7 +814,7 @@ def test_roots_genlaguerre():
     assert_allclose(m, muI, rtol=muI_err)
 
     assert_raises(ValueError, sc.roots_genlaguerre, 0, 2)
-    assert_raises(ValueError, sc.roots_genlaguerre, 3.3, 2)
+    assert_raises(ValueError, sc.roots_genlaguerre, 3.3, 2)  # type: ignore[call-overload]
     assert_raises(ValueError, sc.roots_genlaguerre, 3, -1.1)
 
 

